main.js

```diff
@@ -56,27 +57,24 @@ let appShutdownTimeout;
 
 global.indexUrl = 'file://' + __dirname + '/index.html';
 
-
 function openDevTools() {
   childWindow.webContents.openDevTools({ mode: 'undocked' });
   mainWindow.webContents.openDevTools({ mode: 'undocked' });
 }
 
-// Lazy require OBS
-let _obs;
-
-function getObs() {
-  if (!_obs) {
-    _obs = require('obs-studio-node').NodeObs;
-  }
-
-  return _obs;
-}
-
-
 function startApp() {
   const isDevMode = (process.env.NODE_ENV !== 'production') && (process.env.NODE_ENV !== 'test');
 
+  { // Initialize obs-studio-server
+    // Set up environment variables for IPC.
+    process.env.SLOBS_IPC_PATH = "slobs-".concat(uuid());
+    process.env.SLOBS_IPC_WORKING_DIRECTORY = path.join(app.getAppPath().replace('app.asar', 'app.asar.unpacked') + '/node_modules/obs-studio-node');
+    process.env.SLOBS_IPC_USERDATA = app.getPath('userData');
+    // Host a new IPC Server and connect to it.
+    obs.IPC.ConnectOrHost(process.env.SLOBS_IPC_PATH);
```

Comment:
Does this mean that the main process still maintains an OBS connection?  Are we eventually trying to move away from that?

Reply:
Ideally yes, but the main process is the only one that seems to be able to survive an uncaught v8 exception. All renderer processes basically freeze when they have an uncaught exception, so I wasn't able to have code execute after it.

Reply:
Also it is the one running the crash handler, so it can safely shut down the server if that happens. Except if the main process crashes that is, then it can't.